### ISSUE TITLE
MS2-Spectrum annotation on Mac

### ISSUE THREAD

**Issue by straussmaximilian** [MannLabs/alphaviz#4]
Hi,

I have been trying alphaviz a bit more. I was able to run alphaviz on an M1 Mac with macOS Monterey. ✅
When checking the MS2 spectrum annotation I noticed that there are some differences between different operating systems - it seems that the ion annotation is not present on Mac:

Windows
![Screen Shot 2022-01-31 at 1 51 16 PM](https://user-images.githubusercontent.com/16521288/151796806-c21c0a97-922d-44c1-8916-2db33512e7d3.png)

Mac
![Screen Shot 2022-01-31 at 1 52 31 PM](https://user-images.githubusercontent.com/16521288/151796904-ee276530-42b9-4d39-8906-df8362eb2edd.png)

For another Sequence, ther is some annotation:

Windows
![Screen Shot 2022-01-31 at 1 55 50 PM](https://user-images.githubusercontent.com/16521288/151797446-9be99b2f-12e6-472e-9427-4f9cc0a42ce0.png)

Mac
![Screen Shot 2022-01-31 at 1 56 20 PM](https://user-images.githubusercontent.com/16521288/151797475-4629b71d-4366-4d8a-b7fc-2b9132ba9a8b.png)

When looking at the masses of the ion that is displayed they are slightly different - so maybe the reason for the difference is the missing bruker libraries?

Windows
![Screen Shot 2022-01-31 at 2 00 23 PM](https://user-images.githubusercontent.com/16521288/151798410-dde09c8e-f0a8-4c4a-9080-84a9e2f3c546.png)
Mac
![Screen Shot 2022-01-31 at 2 00 11 PM](https://user-images.githubusercontent.com/16521288/151798412-f296459d-a1d5-429e-8877-2518d9d030cc.png)

I am assuming that the hits and mass deviation are taking from the `msms.txt` from MaxQuant, so I wouldn't have a good clue whats happening here :) 

On another side Note: Would it be possible to link the x-axis of the Intensity plot to the error plot (i.e. that the ion hits are displayed at the same mz position as the error?)

**Comment by swillems** [comment]
This is almost certainly due to missing libraries/calibration indeed. In your first two plots for instance, you can clearly see that the peak at 200 mz is massively shifted. Any chance you used our old test samples that also failed the alphatims integration into alphapept;)

**Comment by straussmaximilian** [comment]
Yes - this is the old test sample with the large calibration offset. 
I checked the `msms.txt`, and for the `y4` it reports `550.2507343470996`, so there is an additional offset probably due to the MaxQuant calibration. 

For the display, one could use the fragment masses from the `msms.txt` so they would always show up. I guess the question here would be if one wants to be closer in showing the raw data or being able to troubleshoot the search engine results. In any case, I think a note for this plot indicating potential differences could be useful.

**Comment by EugeniaVoytik** [comment]
Hi Max,

Thank you very much for submitting this issue. I have reviewed the spectrum annotation process and fixed a small bug there. 😊 As Sander already commented, it's due to lack of libraries on macOS. So the best solution is to either work directly on Windows, or do a conversion to a .hdf file (it's only a minute or so in the AlphaTims CLI). You can then easily use those .hdf files on your macOS. 

I fully agree that there is no problem in showing the ions reported by MQ (I think we could add this optionally if it might be useful for you), but for now the idea is to show if there is evidence of ions found by MQ in the raw data. To make it clearer to users, I will describe this in more detail in the next version of the GUI user guide.

By the way, I've released a new version (v.1.0.7) and have linked the x-axis of the MS spectrum and error plot together for convenience. Feel free to try it! 🙏🏻

![image](https://user-images.githubusercontent.com/33324675/152951914-6b408933-a753-4974-a445-da2bc5711175.png)

Best
Jane

**Comment by straussmaximilian** [comment]
Sounds good to me.
What I would find useful would be to show the raw data, the calibrated data (after calibration from the search engine, not the raw import), and the hits from the search engine. This would allow us to narrow down what is going on: Is the peak there? Is it a search tolerance issue? Is it coming from calibration? But I guess one is limited here as for MaxQuant we can't see how the data was calibrated, so we don't have this information at hand.

For the current plot: I see that you are showing if there is evidence of ions found by MQ, but aren't you comparing uncalibrated to calibrated values here? Wouldn't it make more sense to compare this to the theoretical fragment mass?
E.g., suppose you would have a perfectly calibrated machine and record a spectrum and expect an ion at 100, MaxQuant did a (wrong) calibration, moved it to 100.1, and you have an additional peak at 100.99; this would be highlighted in the plot instead of the one at 100? I would assume the search engine found this, but it actually never did look at this peak. Or maybe I am missing something?

I will check out version 1.07! Thanks.
